### ISSUE TITLE
Real Depth values from <<mono_640x192>> for compute errors!?

### ISSUE THREAD

**Issue by VictorCVision** [nianticlabs/monodepth2#105]
Hi, I want to compute errors for a KITTI random frame (gt, pred). So, I would like to know how to get Real Depth values for prediction map, I have read some questions about to get real depth values from a disparity map using <<mono_640x192>> model, I read that --pred_depth_scale_factor uses a different value for each test image. So...

1.- Can I know the scale factor value if I know Ground Truth image for each frame?

2.- If it is not possible to know scale factor for each frame... Do I have to uses <<0.093>> for multiply my prediction map like in your experiment (Table 8) in the paper?

I am a little confused on how to process the prediction map. What would be the most appropriate way to achieve the evaluation of an image?

****************************************************************************
I have tried to modify some things but most have not given me good results. The following is the best I have obtained.
****************************************************************************

Code for compute and processing prediction map:

`disp = outputs[("disp", 0)]`
`disp_resized = torch.nn.functional.interpolate(disp, (original_height, original_width), mode="bilinear", align_corners=False)`
`pred = (80 * 0.093) / (2 * disp_resized.cpu().numpy()[0][0])`

Obviously I get mask for valid pixels:

`mask = gt > 0`
`gt_valid = gt[mask]`
`pred_valid = pred[mask]`

I have the Ground Truth like:
***** gt_valid *****
 * Shape: (370, 1226)
 * Max value: 81.109375
 * Min value: 2.7578125

for prediction map I have:  

***** pred_valid *****
 * Shape: (370, 1226)
 * Max value: 88.12223
 * Min value: 4.3983316

When a compare random values between gt_valid and pred_valid:

*****
pred_valid[random] = 8.287608
gt[random] = 4.64453125
*****
pred_valid[random] = 42.68488
gt[random] = 35.80078125
*****
pred_valid[random] = 18.136646
gt[random] = 12.6484375
*****
pred_valid[random] = 6.963208
gt[random] = 4.7890625
*****

I get following error values:

![imagen](https://user-images.githubusercontent.com/57361777/68685029-1741a000-052f-11ea-8bf6-bb5e4bdc0546.png)

I hope you can help me. Thank you.

**Comment by mrharicot** [comment]
Hi,
Please have a look at [`evaluate_depth.py`](https://github.com/nianticlabs/monodepth2/blob/master/evaluate_depth.py), you easily be able to edit the code to run on a single image as well as getting the scale value.

**Comment by YunYang1994** [comment]
> Hi,
> Please have a look at [`evaluate_depth.py`](https://github.com/nianticlabs/monodepth2/blob/master/evaluate_depth.py), you easily be able to edit the code to run on a single image as well as getting the scale value.

Can you give us some hint? I am confused too. Thank you in advance !
